### ISSUE TITLE
Possibility to edit changed levels by other levels in Mega Man X3.

### ISSUE THREAD

**Issue by sharivan** [rbrummett/megaedx_v1.3#2]
Hello.

I came across another problem when using this editor, some levels will have its modifing reseted if they are changed after completing another level, for example, when you complete Catfish stage and then Buffalo stage is changed, making any changes in Buffalo stage be reseted. Also, i noticed that changes in Doppler II stage only applies when its boss is Volt Kurageil but not when its boss is Vile, that is, when you modify its stage, the modification will only appears when you defeat Vile in secret area using his weakeness.

Please dont let this project die, Mega Man X's Saga needs more rom hacks.

Thks in advance for listening;

**Comment by rbrummett** [comment]
Hi, thanks for testing the code. I understand the Doppler Stage 2 bug, but what about the other one? I believe there might be a language barrier. You said if you edit one level and save, then edit another level, the first level's changes ("mods") return to normal and are discarded? Meaning you can only edit one level?? I will have to test on a Mega Man X3 rom. Hopefully, this is not a problem on X and X2 also.

**Comment by sharivan** [comment]
Im sorry if i wasn't very clear with my previous explanation, i really dont speak or write english fluently and sometimes i need a translator to translate very long and complex texts.

Trying to explain again about Buffalo's stage, in Mega Man X3, when you complete some stages, other stages will be changed (like in X1). Completing Catfish's stage, Buffalo's stage will be modified having lights in the underground area. The same thing happens when you complete Hornet's stage, modifying Beatle's stage. If you edit Buffalo's stage using MegaED X, these changes will be present when you enter in Buffalo's stage before complete Catfish's stage, buf if you complete Catfish's stage before, when you enter in Buffalo's stage, all editings will be not more present.

This problem happens because some stages when are modified by another, its layout will be obtained not only from the same memory location, but from two memory locations, one for the whole unmodified stage and other for modified version. You can see it when you change the numLevels to 15 or 16 in the line 117 from file MMXCore.cpp and run the program. When you select level 15 for example, you will see the layout for modified Buffalo's stage, but containing alot of glitchy tiles (likely due some incorrect palete reads).

**Comment by rbrummett** [comment]
Yes, the editor always loads specific tiles from the layout. I've tried hacking rom so it changes the conditions (such as Vile still alive for X3) and then loading it in the editor, but it ignores them. I'll try looking at the layout and seeing if there's a way to load different tiles/blocks/etc based off certain flag/conditions
